databases/week3/week3-hw.sql

```diff
@@ -0,0 +1,137 @@
+--Get a meal with any id, fx 1
+SELECT * FROM meal where id=10;
+--Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE  meal SET title = "breakfast buffet", description = "idli, masala dosa, vada, chutney, sambar" WHERE id=10;
+--Delete a meal with any id, fx 1
+DELETE from meal WHERE id=10;
+--Reservation
+--Get all reservations
+SELECT * from reservation;
+--Add a new reservation
+INSERT INTO reservation (number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email)
+VALUES (30, 2, "2021-10-05", "45-99999999", "msk", "msk@mymail.com");
+--Get a reservation with any id, fx 1
+SELECT * FROM reservation WHERE id=9;
+--Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE reservation SET contact_name = "sunil Kumar Mahajan", number_of_guests = 10 WHERE id=9;
+--Delete a reservation with any id, fx 1
+DELETE FROM reservation WHERE id=9;
+--Review
+--Get all reviews
+SELECT * FROM review;
+--Add a new review
+INSERT INTO review (title, description, meal_id, stars, created_date)
+VALUES ("average", "average.. needs improve taste", 7,1, "2021-10-04");
+--Get a review with any id, fx 1
+SELECT * FROM review WHERE id=7;
+--Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE review SET title = "amazing", description = "amazing...." WHERE id = 5;
+--Delete a review with any id, fx 1
+DELETE FROM review WHERE id = 7;
+--  Additional queries
+--Now add a couple of different meals, reservations and reviews with different attributes. With those meals create the following queries
+--Get meals that has a price smaller than a specific price fx 90
+SELECT title, description,price from meal WHERE price <90;
+--Get meals that still has available reservations
+SELECT meal.id AS MEAL_ID, reservation.meal_id AS RESERVED_MEAL_ID, 
+meal.title AS TITLE, meal.max_reservations AS MAX_AVAILABILITY, reservation.number_of_guests AS RESERVED,
+(meal.max_reservations - reservation.number_of_guests) AS STILL_AVAILABLE
+from meal, reservation 
+WHERE meal.id=reservation.meal_id AND meal.created_date = reservation.created_date
+AND meal.max_reservations > (SELECT SUM(reservation.number_of_guests));
+--Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+--*here i am taking partial title as veg*
+SELECT id, title, description FROM meal WHERE title LIKE "%veg%";
+--Get meals that has been created between two dates
+SELECT * FROM meal 
+WHERE created_date BETWEEN "2021-10-05" AND "2021-10-08";
+--Get only specific number of meals fx return only 5 meals
+SELECT * FROM meal LIMIT 5;
+--Get the meals that have good reviews
+SELECT  meal.id, meal.title, review.title, review.stars FROM meal, review
+where  review.stars>4  AND meal.id=review.meal_id;
+--Get reservations for a specific meal sorted by created_date
+SELECT meal.id, reservation.meal_id, meal.title, 
```

Comment:
Why you need to test **meal.title** ? Can you  modify it like this and see what it gives out
```
SELECT meal.id, reservation.meal_id, meal.title, 
reservation.number_of_guests, reservation.created_date FROM meal, reservation
WHERE  meal.id=reservation.meal_id
ORDER BY reservation.created_date;
```

Reply:
Hi
I tried your query, but it gives me all the meals order by created date. But the task is to get specific meal . "Get reservations for a **_specific meal_** sorted by created_date".  Please correct me if i am wrong.

Reply:
--Get reservations for a specific meal sorted by created_date?
Hi. I execute the above query it will display all the meals order by created date, But the task is to display only specific meal  sorted by created_date" .For that I took meal title.